app/src/main/java/org/oppia/android/app/options/TextSizeItemViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package org.oppia.android.app.options
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import org.oppia.android.app.model.ReadingTextSize
+import org.oppia.android.app.viewmodel.ObservableViewModel
+
+private const val SMALL_TEXT_SIZE_SCALE = 0.8f
+private const val MEDIUM_TEXT_SIZE_SCALE = 1.0f
+private const val LARGE_TEXT_SIZE_SCALE = 1.2f
+private const val EXTRA_LARGE_TEXT_SIZE_SCALE = 1.4f
+
+/**
+ * Text Size item view model for the recycler view in [ReadingTextSizeFragment].
+ */
+class TextSizeItemViewModel constructor(
+  val defaultReadingTextSizeInFloat: Float,
+  val readingTextSize: ReadingTextSize,
+  private val selectedTextSize: LiveData<String>,
+  val textSizeRadioButtonListener: TextSizeRadioButtonListener
+) : ObservableViewModel() {
+  val textSizeName: String by lazy {
+    when (readingTextSize) {
+      ReadingTextSize.SMALL_TEXT_SIZE -> "Small"
```

Comment:
All these strings should be defined in `strings.xml` file.

Reply:
done